Sources/StreamChat/Workers/EventNotificationCenter.swift

```diff
@@ -64,6 +103,171 @@ class EventNotificationCenter: NotificationCenter, @unchecked Sendable {
             }
         })
     }
+
+    private func convertManualEventsToDomain(_ events: [Event]) -> [Event] {
+        events.compactMap { event in
+            guard let eventDTO = event as? EventDTO else {
+                return nil
+            }
+
+            let eventPayload = eventDTO.payload
+
+            guard let cid = eventPayload.cid else {
+                return nil
+            }
+
+            switch eventPayload.eventType {
+            case .messageNew:
+                return createMessageNewEvent(from: eventPayload, cid: cid)
+                
+            case .messageUpdated:
+                return createMessageUpdatedEvent(from: eventPayload, cid: cid)
+                
+            case .messageDeleted:
+                return createMessageDeletedEvent(from: eventPayload, cid: cid)
+                
+            case .reactionNew:
+                return createReactionNewEvent(from: eventPayload, cid: cid)
+                
+            case .reactionUpdated:
+                return createReactionUpdatedEvent(from: eventPayload, cid: cid)
+                
+            case .reactionDeleted:
+                return createReactionDeletedEvent(from: eventPayload, cid: cid)
+                
+            default:
+                return nil
+            }
+        }
+    }
+    
+    // MARK: - Event Creation Helpers
+    
+    private func createMessageNewEvent(from payload: EventPayload, cid: ChannelId) -> MessageNewEvent? {
+        guard
+            let userPayload = payload.user,
+            let messagePayload = payload.message,
+            let createdAt = payload.createdAt,
+            let channel = try? database.writableContext.channel(cid: cid)?.asModel(),
```

Comment:
how does this exactly work? We won't have the channel in the db, right?

Reply:
The problem here is that our domain events require a channel model, and most of the events don't include the channel data. So I'm basically fetching the channel, just to be able to create the event, but we don't really need it. I will probably cache this data as well.

Reply:
Added the chaching 👍